src/serverenvironment.cpp

```diff
@@ -974,6 +983,39 @@ class ABMHandler
 					}
 					// No required neighbor found
 					continue;
+				} else if (aabm.check_required_neighbors || aabm.check_without_neighbors) {
```

Comment:
Can we integrate this all in the first loop?

Reply:
Yes, it can be integrated into one loop by using more conditions and the goto command.

I did not do it, because I didn't want to add something to cycle for normal, commonly used ABMs.

So, if you think that it should not be a problem, it is not a problem to use only one loop.